runtime/interpreter/value.go

```diff
@@ -7147,3 +7149,130 @@ func (v LinkValue) String() string {
 		v.TargetPath.String(),
 	)
 }
+
+// BuiltinCompositeValue
+type BuiltinCompositeValue struct {
+	Fields     *StringValueOrderedMap
+	staticType StaticType
+	structType *sema.BuiltinCompositeType
+}
+
+func NewBuiltinCompositeValue(structType *sema.BuiltinCompositeType, fields *StringValueOrderedMap) *BuiltinCompositeValue {
+	if fields == nil {
+		fields = NewStringValueOrderedMap()
+	}
+
+	return &BuiltinCompositeValue{
+		Fields:     fields,
+		structType: structType,
+		staticType: ConvertSemaToPrimitiveStaticType(structType),
+	}
+}
+
+func (*BuiltinCompositeValue) IsValue() {}
+
+func (v *BuiltinCompositeValue) Accept(interpreter *Interpreter, visitor Visitor) {
+	visitor.VisitBuiltinCompositeValue(interpreter, v)
+}
+
+func (v *BuiltinCompositeValue) DynamicType(_ *Interpreter) DynamicType {
+	return BuiltinCompositeDynamicType{v.structType}
+}
+
+func (v *BuiltinCompositeValue) StaticType() StaticType {
+	return v.staticType
+}
+
+func (v *BuiltinCompositeValue) Copy() Value {
+	return v
+}
+
+func (*BuiltinCompositeValue) GetOwner() *common.Address {
+	// value is never owned
+	return nil
+}
+
+func (*BuiltinCompositeValue) SetOwner(_ *common.Address) {
+	// NO-OP: value cannot be owned
+}
+
+func (*BuiltinCompositeValue) IsModified() bool {
+	return false
+}
+
+func (*BuiltinCompositeValue) SetModified(_ bool) {
+	// NO-OP
+}
+
+func (v *BuiltinCompositeValue) Destroy(_ *Interpreter, _ LocationRange) trampoline.Trampoline {
+	return trampoline.Done{}
+}
+
+func (v *BuiltinCompositeValue) String() string {
+	return formatComposite(string(v.structType.ID()), v.Fields)
+}
+
+func (v *BuiltinCompositeValue) GetMember(_ *Interpreter, _ LocationRange, name string) Value {
+	value, _ := v.Fields.Get(name)
+	return value
+}
+
+func (*BuiltinCompositeValue) SetMember(_ *Interpreter, _ LocationRange, _ string, _ Value) {
+	panic(errors.NewUnreachableError())
+}
+
+// NewAccountKeyValue constructs an AccountKey value.
+func NewAccountKeyValue(
```

Comment:
Nice! Can you please open an issue for refactoring the other built-in values to this new built-in composite value, too?

Reply:
Created https://github.com/onflow/cadence/issues/650